### PR TITLE
Glharper/conversation transcriber

### PR DESCRIPTION
**Pull request by glharper**
Add conversation transcriber for handling transcription of conversation from enrolled users

### CODE REVIEW

File: tests/ConversationTranscriberTests.ts [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                        });
+                },
+                (error: string) => {
+                    done.fail(error);
+                });
+        },
+        (error: string) => {
+            done.fail(error);
+        });
+});
```
[the remark this answers is not in the document]
Comment (glharper) [reply tests/ConversationTranscriberTests.ts:348]: I will add a task to get add this, and fix whatever other disconnect tests currently aren't working.

File: tests/ConversationTranscriberTests.ts [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    const s: sdk.SpeechTranslationConfig = BuildSpeechConfig();
+    objsToClose.push(s);
+
+    const c: sdk.Conversation = CreateConversation(s);
+    objsToClose.push(c);
+
+    const t: sdk.ConversationTranscriber = BuildTranscriber();
+    t.sessionStopped = (o: sdk.ConversationTranscriber, e: sdk.SessionEventArgs) => {
+        try {
+            done();
```
[the remark this answers is not in the document]
Comment (glharper) [reply tests/ConversationTranscriberTests.ts:164]: When an expect fails, it fails the test, even if the done is reached, as long as the fail is in a try/catch block.

File: tests/ConversationTranscriberTests.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (glharper) [reply tests/ConversationTranscriberTests.ts:388]: https://msasg.visualstudio.com/Skyman/_workitems/edit/2957459

File: tests/ConversationTranscriberTests.ts [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+                        () => {
+                            expect(c.participants).not.toBeUndefined();
+                            expect(c.participants.length).toEqual(2);
+                            t.canceled = (o: sdk.ConversationTranscriber, e: sdk.ConversationTranscriptionCanceledEventArgs) => {
+                                try {
+                                    expect(e.errorDetails).toBeUndefined();
+                                    expect(e.reason).toEqual(sdk.CancellationReason.EndOfStream);
+                                } catch (error) {
+                                    done.fail(error);
+                                }
```
[the remark this answers is not in the document]
Comment (glharper) [reply tests/ConversationTranscriberTests.ts]: Changed to define immediately after transcriber instantiation

File: tests/ConversationTranscriberTests.ts [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+    t.sessionStopped = (o: sdk.ConversationTranscriber, e: sdk.SessionEventArgs) => {
+        try {
+            done();
+        } catch (error) {
+            done.fail(error);
+        }
+    };
+    t.joinConversationAsync(c,
+        () => {
+            expect(t.properties).not.toBeUndefined();
```
[the remark this answers is not in the document]
Comment (glharper) [reply tests/ConversationTranscriberTests.ts]: Added.

File: src/sdk/Transcription/TranscriberRecognizer.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+    public startContinuousRecognitionAsync(cb?: () => void, err?: (e: string) => void): void {
+        marshalPromiseToCallbacks(this.startContinuousRecognitionAsyncImpl(RecognitionMode.Conversation), cb, err);
+    }
+
+    public stopContinuousRecognitionAsync(cb?: () => void, err?: (e: string) => void): void {
+        marshalPromiseToCallbacks(this.stopContinuousRecognitionAsyncImpl(), cb, err);
+    }
+
+    public async close(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (rhurey) [reply src/sdk/Transcription/TranscriberRecognizer.ts]: Ah. I think this being in the SDK directory confused me.

---
In reply to: [474992179](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/226#discussion_r474992179) [](ancestors = 474992179)

**Pull request merged**